docs/PersistedState.md

```diff
@@ -0,0 +1,93 @@
+# Persisted State Architecture
+
+For persisted state, we are migrating from using Zustand's [global store](https://zustand.docs.pmnd.rs/integrations/persisting-store-dat) to using [Zustand with React Context](https://tkdodo.eu/blog/zustand-and-react-context). This change improves testability by allowing us to inject temporary stores in place of persisted state, making it easier to manipulate data and perform isolated tests.
+
+Following this pattern:
+
+```tsx
+import {createContext, useContext} from 'react';
+import {createStore, useStore, type StoreApi} from 'zustand';
+import {persist as createPersistedState} from 'zustand/middleware';
+
+type BearState = {bears: number};
+
+function createInitialState(): BearState {
+  return {bears: 0};
+}
+
+const BEAR_STORE_PERSIST_KEY = 'bearStore';
+
+export function createBearStore({persist = false}: {persist: boolean}) {
+  let instance: StoreApi<BearState>;
+
+  if (persist) {
+    instance = createStore(
+      createPersistedState(createInitialState, {name: BEAR_STORE_PERSIST_KEY}),
+    );
+  } else {
+    instance = createStore(createInitialState);
+  }
+
+  const actions = {
+    increasePopulation() {
+      instance.setState(state => ({bears: state.bears + 1}));
+    },
+    removeAllBears: () => instance.setState({bears: 0}),
+    updateBears: (newBears: number) => instance.setState({bears: newBears}),
+  };
+
+  return {instance, actions};
+}
+
+const BearStoreContext = createContext<ReturnType<
+  typeof createBearStore
+> | null>(null);
+
+export const BearStoreProvider = ({instance, children}) => {
+  return (
+    <BearStoreContext.Provider value={instance}>
+      {children}
+    </BearStoreContext.Provider>
+  );
+};
+
+// Shouldn't need to export this, just a helper for DRY
+function useBearContext() {
+  const value = useContext(BearStoreContext);
+  if (!value) {
+    throw new Error('Must set up the BearStoreProvider first');
+  }
+  return value;
+}
+
+export function useBearState(): BearState;
+export function useBearState<T>(selector: (state: BearState) => T): T;
+export function useBearState<T>(selector?: (state: BearState) => T) {
+  const {store} = useBearContext();
+  return useStore(store, selector!);
+}
+
+// No need for a selector because this is stable across renders
+export function useBearActions() {
+  const {actions} = useBearContext();
+  return actions;
+}
+```
+
+Main things to note:
+
+- We are returning the store state as `instance` and the actions that manipulate the state as `actions`
```

Comment:
Naming nit: We are returning a store _instance_ as `instance`, but in the hooks we return the store _state_ (as returned from `storeInstance.getState()`)